src/SvelteAsset.js

```diff
@@ -8,28 +9,43 @@ class SvelteAsset extends Asset {
   }
 
   async generate() {
-    let svelteOptions = {
-      compilerOptions: {
-        generate: 'dom',
-        format: 'cjs',
-        store: true,
-        filename: this.relativeName,
-        css: false
-      },
-      preprocess: undefined
+    let compilerOptions = {
+      generate: 'dom',
+      format: 'cjs',
+      store: true,
+      css: false
+    };
+    let preprocessOptions = undefined;
+
+    const fixedCompilerOptions = {
+      filename: this.relativeName,
+      // the name of the constructor. Required for 'iife' and 'umd' output,
+      // but otherwise mostly useful for debugging. Defaults to 'SvelteComponent'
+      name: capitalize(sanitize(this.relativeName)) 
     };
 
     const customConfig = await this.getConfig(['.svelterc', 'svelte.config.js', 'package.json']);
     if (customConfig) {
-      svelteOptions = Object.assign(svelteOptions, customConfig.svelte || customConfig);
+      const customOptions = customConfig.svelte || customConfig;
+      if (customOptions.preprocess) {
+        preprocessOptions = customOptions.preprocess;
+      }
+      if (customOptions.compilerOptions) {
+        Object.keys(customOptions.compilerOptions).forEach(key => {
+          if (fixedCompilerOptions[key]) return;
+          compilerOptions[key] = customOptions.compilerOptions[key];
+        });
+      }
     }
 
-    if (svelteOptions.preprocess) {
-      const preprocessed = await preprocess(this.contents, svelteOptions.preprocess);
+    compilerOptions = Object.assign({}, compilerOptions, fixedCompilerOptions);
```

Comment:
Why not simply do Object.assign(customOptions.compilerOptions || {}, compilerOptions, fixedCompilerOptions);